R/plot.R

```diff
@@ -0,0 +1,111 @@
+
+  params_and_times[, mu := biokinetics_simulate_trajectory(t, t0, tp, ts, m1, m2, m3),
+                     by = c("t", "t0", "tp", "ts", "m1", "m2", "m3")]
+
+  summary <- params_and_times[, .(me = stats::quantile(mu, 0.5, names = FALSE),
+                                  lo = stats::quantile(mu, 0.025, names = FALSE),
+                                  hi = stats::quantile(mu, 0.975, names = FALSE)), by = t]
+
+  plot <- ggplot(summary) +
+    geom_line(aes(x = t, y = me)) +
+    geom_ribbon(aes(x = t, ymin = lo, ymax = hi), alpha = 0.5)
+
+  if (!is.null(data)) {
+    plot <- plot + geom_point(data = data, aes(x = time_since_last_exp, y = value))
+  }
+  plot
+}
+
+plot_data <- function(data, covariates) {
+  validate_required_cols(data, c("time_since_last_exp", "value", "titre_type"))
+  # Declare variables to suppress notes when compiling package
+  # https://github.com/Rdatatable/data.table/issues/850#issuecomment-259466153
+  time_since_last_exp <- value <- titre_type <- NULL
+
+  ggplot(data) +
+    geom_point(aes(x = time_since_last_exp, y = value, colour = titre_type)) +
+    geom_smooth(aes(x = time_since_last_exp, y = value, colour = titre_type)) +
+    facet_wrap(eval(parse(text = facet_formula(covariates)))) +
+    guides(colour = guide_legend(title = "Titre type"))
+}
+
+#' Plot method for "biokinetics_population_trajectories" class
+#'
+#' @param x An object of class "biokinetics_population_trajectories". These are
+#' generated by running biokinetics$simulate_populate_trajectories(). See
+#' \href{../../epikinetics/html/biokinetics.html#method-biokinetics-simulate_population_trajectories}{\code{biokinetics$simulate_population_trajectories()}}
+#' @param \dots Further arguments passed to the method.
+#' @param data Optional data.table containing raw data as provided to the biokinetics model.
+#' @export
+plot.biokinetics_population_trajectories <- function(x, ..., data = NULL) {
+  covariates <- attr(x, "covariates")
+
+  # Declare variables to suppress notes when compiling package
+  # https://github.com/Rdatatable/data.table/issues/850#issuecomment-259466153
+  time_since_last_exp <- value <- me <- titre_type <- lo <- hi <- NULL
+  day <- last_exp_day <- NULL
+
+  if (attr(x, "summarised")) {
+    plot <- ggplot(x) +
+      geom_line(aes(x = time_since_last_exp, y = me, colour = titre_type)) +
+      geom_ribbon(aes(x = time_since_last_exp,
+                      ymin = lo,
+                      ymax = hi,
+                      fill = titre_type), alpha = 0.5)
+  } else {
+    plot <- ggplot(x) +
+      geom_line(aes(x = time_since_last_exp, y = mu, colour = titre_type), alpha = 0.5)
+  }
+
+  if (!is.null(data)) {
+    validate_required_cols(data)
+    plot <- plot +
+      geom_point(data = data,
+                 aes(x = as.integer(day - last_exp_day, units = "days"),
+                     y = value), size = 0.5, alpha = 0.5)
+  }
+  plot +
+    scale_y_continuous(trans = "log2") +
```

Comment:
I thought the output of simulate_population_trajectories was already on the log2 scale? why does this need to be transformed here?

Reply:
The trajectories are always returned on the original scale (so log if the input data was on the log scale, but back-transformed to natural scale if the inputs were natural scale). Does this make sense? We could change this behaviour if confusing.

Reply:
yeah makes sense, I think default behaviour just needs to be explicit (which is assuming that data is on a natural scale right)?

Reply:
ah good catch this should be conditional on the input data being on a natural scale!